token.go

```diff
@@ -0,0 +1,53 @@
+package profitbricks
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"strings"
+)
+
+type tokenHeader struct {
+	Typ string
+	Kid string
+	Alg string
+}
+
+// ExtractIDFromToken returns the given token's key ID
+func ExtractIDFromToken(token string) (string, error) {
+	headerB64 := strings.Split(token, ".")[0]
+	headerMarshaled, err := base64.StdEncoding.DecodeString(headerB64)
+	if err != nil {
+		return "", err
+	}
+	var header tokenHeader
+	if err = json.Unmarshal(headerMarshaled, &header); err != nil {
+		return "", err
+	}
+	return header.Kid, nil
+}
+
+// TokenID returns the client's token's key ID if a token is set.
+// Returns an empty string when using basic auth.
+func (c *Client) TokenID() (string, error) {
+	if c.client.token != "" {
+		return ExtractIDFromToken(c.client.token)
+	}
+	return "", nil
+}
+
+// DeleteToken deletes the given token
+func (c *Client) DeleteToken(token string) error {
```

Comment:
DeleteCurrentToken should delete the current (auth) token, DeleteToken should use the authToken for auth and delete the requested token. It would be strange if an admin would use this to delete a token, but the activity log shows that the user deleted the token by himself.